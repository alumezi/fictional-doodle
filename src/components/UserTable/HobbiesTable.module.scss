@import '../../variables.scss';

.hobbiesTable {
    border: 1px solid $border-color;
}

.header {
    height: 5rem;
    border-bottom: 1px solid $border-color;
    text-align: left;
    vertical-align: middle;
    padding: 1rem;
}

.elementWrapper {
    display: flex;
    min-height: 10rem;

    .leftComponent {
        width: 30%;
        margin-right: 1rem;
        resize: horizontal;
        overflow: auto;
        border: 1px solid $border-color;
    }

    .rightComponent {
        width: 69%;
        resize: horizontal;
        overflow: auto;
        border: 1px solid $border-color;
    }
}
