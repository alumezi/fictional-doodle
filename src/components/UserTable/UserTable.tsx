import React, { PropsWithChildren, useState } from 'react'
import { User } from '../../interfaces'
import styles from './HobbiesTable.module.scss'

export function Usertable({
    users,
    addUser,
    onSelect,
}: PropsWithChildren<{
    users: User[]
    addUser: (userName: string) => void
    onSelect: (userID: string) => void
}>) {
    const [newUser, setNewUser] = useState('')

    const handleSetAddUser = () => {
        addUser(newUser)
        setNewUser('')
    }

    if (!users || !users.length) {
        return null
    }

    return (
        <div className={styles.hobbiesTable}>
            <div className={styles.tableRow}>
                <input
                    className={styles.input}
                    aria-label="Enter user name"
                    value={newUser}
                    onChange={(e) => setNewUser(e.target.value)}
                />
                <button onClick={handleSetAddUser}>Add</button>
            </div>
            <div>
                {users.map((user) => (
                    <div
                        className="user"
                        key={user.id}
                        onClick={() => onSelect(user.id)}
                    >
                        {user.name}
                    </div>
                ))}
            </div>
        </div>
    )
}
