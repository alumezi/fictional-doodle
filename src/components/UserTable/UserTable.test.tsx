import React from 'react'
import { shallow } from 'enzyme'
import { Usertable } from './UserTable'

let mockUsers = [{ id: '123ewg43d3d', name: 'Arbnor Lumezi' }]
let wrapper: any
beforeEach(() => {
    wrapper = shallow(
        <Usertable users={mockUsers} addUser={() => {}} onSelect={() => {}} />
    )
})

describe('<Usertable /> rendering', () => {
    it('renders correctly', () => {
        expect(wrapper).toMatchSnapshot()
    })

    it('renders users', () => {
        expect(wrapper.find('.user').text()).toEqual('Arbnor Lumezi')
    })
})
