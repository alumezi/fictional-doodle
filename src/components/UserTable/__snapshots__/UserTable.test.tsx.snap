// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Usertable /> rendering renders correctly 1`] = `ShallowWrapper {}`;
